addons/survey/static/src/js/survey_timer.js

```diff
@@ -24,18 +24,17 @@ publicWidget.registry.SurveyTimerWidget = publicWidget.Widget.extend({
 
 
     /**
-    * Two responsabilities : Validate that time limit is not exceeded and Run timer otherwise.
-    * If end-user's clock OR the system clock  is de-synchronized before the survey is started, we apply the
-    * difference in timer (if time difference is more than 5 seconds) so that we can
-    * display the 'absolute' counter
+    * Two responsibilities: Validate that the time limit is not exceeded and Run timer otherwise.
+    * If the end-user's clock OR the system clock is desynchronized, we apply the difference in the clocks.
+    * This makes the timer fair across users and avoids early submissions to the server.
     *
     * @override
     */
     start: function () {
         var self = this;
         return this._super.apply(this, arguments).then(function () {
             self.countDownDate = moment.utc(self.timer).add(self.timeLimitMinutes, 'minutes');
-            if (Math.abs(self.timeDifference) >= 5000) {
+            if (self.timeDifference) {
```

Comment:
Hello @LightSystem ,

Thanks for the PR!

Your point is valid, we can see how this 5 seconds threshold can be problematic in real use, although we've not received much feedback about it.
As it shouldn't change anything visually for the participant and as we try to limit the changes in stable to the strict necessary, would you mind changing the threshold to `500`ms instead of removing it completely?